app/models/vote.rb

```diff
@@ -0,0 +1,22 @@
+class Vote < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :review
+
+  validates :user, presence: true
+  validates :review, presence: true
+  validates :user_id, uniqueness: {
+    scope: :review_id, message: "User can only vote once per review."
+  }
+
+  def self.user_already_voted(user, review)
+    Vote.where("(user_id = #{user.id} AND review_id = #{review})")
+  end
+
+  def self.destroy_user_vote(user, review)
+    review_id = review.id
+    vote = Vote.where("(user_id = #{user.id} AND review_id = #{review_id})")
+    review = Review.find(review_id)
+    review.update_attributes(vote_value: review.vote_value -= vote.first.value)
+    Vote.find(vote.first.id).destroy
+  end
+end
```

Comment:
I need me some unit tests on this guy! You have two custom methods here, which means you need to test them in your Vote unit tests as well as testing Vote's associations and validations. 

Check out some of [Horizon's specs](https://github.com/LaunchAcademy/event_horizon/blob/master/spec/models/lesson_spec.rb) for examples on how custom methods are tested. Note how each of Lesson's methods has its own `describe` block, and inside there are several tests for that method, possible each inside their own context.